adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -257,9 +258,15 @@ private void performInstanceDiscovery(final URL authorityUrl, final String trust
                         new InstanceDiscoveryMetadata(authorityUrl.getHost(), authorityUrl.getHost(), aliases));
             }
             result = AuthorityValidationMetadataCache.isAuthorityValidated(authorityUrl);
-        } catch (final IOException | JSONException e) {
-            Logger.e(TAG + methodName, "Error when validating authority. ", "", ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_URL, e);
+        } catch (JSONException e) {
+            Logger.e(TAG + methodName, "Error when validating authority. ", "", ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_INSTANCE, e);
             throw new AuthenticationException(ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_INSTANCE, e.getMessage(), e);
+        } catch (SocketTimeoutException e){
+            Logger.e(TAG + methodName, "Error when validating authority. ", "", ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE, e);
+            throw new AuthenticationException(ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE, e.getMessage(), e);
```

Comment:
Let's create an entirely new error for this, such as: `SOCKET_TIMEOUT`

`DEVICE_CONNECTION_IS_NOT_AVAILABLE` is same `AdalError` that will be returned by `HttpWebRequest#throwIfNetworkNotAvailable` -- while these circumstances both indicate some kind of network connectivity trouble, the actual error mode [can be quite different](https://stackoverflow.com/questions/7360520/connectiontimeout-versus-sockettimeout).

In the case of socket timeouts, we want to track (via logging/telemetry) that the device _did_ have a connection, but that it was the Idp/STS who 'dropped' the connection